### PR TITLE
Move to SpeziValidation package

### PR DESCRIPTION
**Pull request by Supereg**
# Move to SpeziValidation package

## :recycle: Current situation & Problem
Currently, SpeziAccount contains a lot of code that is not specific to account management. One part is input validation. https://github.com/StanfordSpezi/SpeziViews/pull/21 introduces the new `SpeziValidation` library, based on previous code in SpeziAccount. This PR makes the efforts to move to that new software package and greatly simplify the `SpeziAccount` code base.

## :gear: Release Notes 
* Update to `SpeziValidation`, a new one-stop software package for input validation

## :books: Documentation
Documentation was addressed where needed.

_TBA_: Some documentation still requires external links once the SpeziViews PR is merged.

## :white_check_mark: Testing
Tests were adapted where needed. Nothing new was added.

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Supereg** [comment]
This PR still misses some external link to the latest SpeziViews documentation once https://github.com/StanfordSpezi/SpeziViews/pull/21 is merged and tagged. And of course updating the Package.swift file to the new release version.
Other than that, the PR is ready to review.

File: Sources/SpeziAccount/AccountService/Configuration/FieldValidationRules.swift [hunk 1]
```diff
@@ -7,15 +7,18 @@
 //
 
 import Spezi
+import SpeziValidation
 
+// TODO: add external documentation links
```
Comment (PSchmiedmayer) [inline Sources/SpeziAccount/AccountService/Configuration/FieldValidationRules.swift:12]: Small TODO to address the missing documentation.

File: Sources/SpeziAccount/AccountValue/AccountKey.swift [hunk 2]
```diff
@@ -47,8 +47,8 @@ public protocol AccountKey: KnowledgeSource<AccountAnchor> where Value: Sendable
     /// The ``DataEntryView`` that is used to enter a value for this account value.
     ///
     /// This view is wrapped into a ``GeneralizedDataEntryView`` and used in views like the ``SignupForm`` to enter the account value.
-    /// For example, for a String-based account value, one might define a ``DataEntryView`` based on `TextField` or ``VerifiableTextField``.
-    associatedtype DataEntry: DataEntryView<Self>
+    /// For example, for a String-based account value, one might define a ``DataEntryView`` based on `TextField` or `VerifiableTextField`.
+    associatedtype DataEntry: DataEntryView<Self> // TODO: link to SpeziValidation
```
Comment (PSchmiedmayer) [inline Sources/SpeziAccount/AccountValue/AccountKey.swift:51]: Comment to ensure that we don't forget about this TODO here.

File: Sources/SpeziAccount/AccountValue/Keys/PersonNameKey.swift [hunk 3]
```diff
@@ -6,10 +6,14 @@
 // SPDX-License-Identifier: MIT
 //
 
+import SpeziPersonalInfo
+import SpeziValidation
 import SpeziViews
 import SwiftUI
 
 
+// TODO test compile documentation
```
Comment (PSchmiedmayer) [inline Sources/SpeziAccount/AccountValue/Keys/PersonNameKey.swift:15]: Comment that we don't forget about the TODO here.

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for all the changes and great to see Spezi Validation in action!

Looked over the PR and it looks great. Addressing the last PRs and pulling in the latest version of SpeziViews sounds good and I am happy to see the PR being merged 🚀

**Comment by PSchmiedmayer** [comment]
Thank you @Supereg! 🚀

**Pull request merged**